include/nbl/ext/DebugDraw/CDraw3DLine.h

```diff
@@ -10,131 +10,189 @@
+            const core::vectorSIMDf& fromColor  = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f),
+            const core::vectorSIMDf& toColor    = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f)
+          ) noexcept
+          {
+            m_viewProj = viewProj;
+
+            m_lines = S3DLineVertexPairList{
+              std::pair(
+                S3DLineVertex{{ from.x, from.y, from.z }, { fromColor.r, fromColor.g, fromColor.b, fromColor.a }},
+                S3DLineVertex{{ to.x, to.y, to.z }, { toColor.r, toColor.g, toColor.b, toColor.a }}
+              )
+            };
+          }
+
+          inline void clearData() noexcept { m_lines.clear(); }
+
+          inline void addLine(
+            const core::matrix4SIMD& viewProj,
+            const core::vectorSIMDf& from, const core::vectorSIMDf& to,
+            const core::vectorSIMDf& fromColor  = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f),
+            const core::vectorSIMDf& toColor    = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f)
+          ) noexcept
+          {
+            m_viewProj = viewProj;
+
+            m_lines.emplace_back(
+              S3DLineVertex{{ from.x, from.y, from.z }, { fromColor.r, fromColor.g, fromColor.b, fromColor.a }},
+              S3DLineVertex{{ to.x, to.y, to.z }, { toColor.r, toColor.g, toColor.b, toColor.a }}
+            );
+          }
+
+          inline void addLine(
+            const core::vectorSIMDf& from, const core::vectorSIMDf& to,
+            const core::vectorSIMDf& fromColor  = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f),
+            const core::vectorSIMDf& toColor    = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f)
+          ) noexcept
+          {
+            m_lines.emplace_back(
+              S3DLineVertex{{ from.x, from.y, from.z }, { fromColor.r, fromColor.g, fromColor.b, fromColor.a }},
+              S3DLineVertex{{ to.x, to.y, to.z }, { toColor.r, toColor.g, toColor.b, toColor.a }}
+            );
+          }
+
+          inline void addLines(const S3DLineVertexPairList& linesData) noexcept
+          { m_lines.insert(m_lines.end(), linesData.begin(), linesData.end()); }
+
+          // to use for both AABB and OBB (cannot/should not alter bbox here!)
+          template<class TBox = core::aabbox3dsf>
+          inline void addBox(
+            const TBox& box,
+            const core::matrix3x4SIMD& tform    = core::matrix3x4SIMD(),
+            const core::vectorSIMDf& fromColor  = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f),
+            const core::vectorSIMDf& toColor    = core::vectorSIMDf(0.f, 1.f, 0.f, 1.f)
+          ) noexcept
+          {
+            static_assert(
+              std::is_base_of<core::aabbox3dsf, TBox>::value ||
+              std::is_base_of<core::aabbox3df,  TBox>::value ||
+              std::is_base_of<core::aabbox3di,  TBox>::value,
+              "TBox should derive from aabbox3d<T, UVector>"
+            );
+
+            core::vectorSIMDf verts[10];
+
+            box.getEdges(verts);
```

Comment:
this is horrible method, neither the AABB or the OBB should have it inside them, it inverts the ordering of logic (object should not know or care about metadata which is derived from it)

I know its present in the AABB, and this is likely @Przemog1's code, but its quite wrong

Reply:
I think, this method can be re-implemented in a way that for both aabb and obb, it outputs the side/edge vectors from the matrix instead of using center point. IMHO, it is not horrible, just needs a little more care :)